config/dependency.go

```diff
@@ -382,28 +386,100 @@ func cloneTerragruntOptionsForDependencyOutput(terragruntOptions *options.Terrag
 	return targetOptions, nil
 }
 
-// runTerragruntOutputJson uses terragrunt running functions to extract the json output from the target config. Make a
-// copy of the terragruntOptions so that we can reuse the same execution environment.
-func runTerragruntOutputJson(terragruntOptions *options.TerragruntOptions, targetConfig string) ([]byte, error) {
-	targetOptions, err := cloneTerragruntOptionsForDependencyOutput(terragruntOptions, targetConfig)
+// Retrieve the outputs from the terraform state in the target configuration. This attempts to optimize the output
+// retrieval if the following conditions are true:
+// - State backends are managed with a `remote_state` block.
```

Comment:
For backends not managed this way, what would we need to do? I guess we'd need to be able to (a) check out the source code, which is either in the current working dir, or checked out based on the `source` param, and (b) run any `generate` blocks. As long as the `source` param and `generate` blocks don't depend on other `dependency` blocks, could we add a special case here too, that checks out the code, runs the code generation, and then runs `output`?